holoviews/operation/element.py

```diff
@@ -758,8 +758,12 @@ def _process(self, p, element, ranges={}):
         for d1, d2 in permuted_dims:
             if d1 == d2:
                 if p.diagonal_type is not None:
-                    values = element.dimension_values(d1)
-                    el = p.diagonal_type(values, vdims=[d1])
+                    if p.diagonal_type._1d:
```

Comment:
Naming this one is hard!

Also, I was wondering if the suggestion for utilities to get these properties might help with figuring out appropriate output elements after reducing the number of dimensions? These utilities could look at something like ``element._properties`` defined at the class level which could be a dictionary (with a key ``integer_index_1d`` - or something similar in this case)?

This helps the naming issue a bit because if you have a function, at least there is a docstring to consult describing what it is, instead of trying to find a sufficiently descriptive attribute name.

Not something we should do now though. From what I understand about it - how about ``_integer_indexable_1d``? Still not great though...